packages/loader/container-loader/src/container.ts

```diff
@@ -1561,6 +1584,10 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                     this.deltaManager
                     : undefined;
 
+            if (this.protocolDetails?.connectionHandler !== undefined) {
+                this.protocolDetails?.connectionHandler(this._protocolHandler, details, opsBehind);
```

Comment:
Why it's not a method on this._protocolHandler itself?

Reply:
Ideally we use existing setConnectionState, but change it's signature (and expectations) to allow tri-states in.